modules/dynamic/src-jvm/DynamicSchemaIndexPlatform.scala

```diff
@@ -16,28 +16,99 @@
 
 package smithy4s.dynamic
 
-import software.amazon.smithy.model.shapes.ModelSerializer
-import software.amazon.smithy.model.transform.ModelTransformer
+import smithy4s.Service
+import smithy4s.schema.OperationSchema
+import software.amazon.smithy.model.shapes.Shape
+import smithy4s.schema.Schema
+import software.amazon.smithy.model.shapes.OperationShape
+import scala.jdk.CollectionConverters._
+import software.amazon.smithy.model.shapes.ServiceShape
+import software.amazon.smithy.model.Model
 
 private[dynamic] trait DynamicSchemaIndexPlatform {
-  self: DynamicSchemaIndex.type =>
+  self: DynamicSchemaIndex =>
+
+  import smithy4s.dynamic.internals.conversion.syntax._
 
   /**
-    * Loads a dynamic schema index model from a smithy model.
+    * Convert the `DynamicSchemaIndex` into a smithy `Model`.
     */
-  def loadModel(
-      model: software.amazon.smithy.model.Model
-  ): DynamicSchemaIndex = {
-    val flattenedModel =
-      ModelTransformer.create().flattenAndRemoveMixins(model);
-    val node = ModelSerializer.builder().build.serialize(flattenedModel)
-    val document = NodeToDocument(node)
-    smithy4s.Document
-      .decode[smithy4s.dynamic.model.Model](document)
-      .map(load(_)) match {
-      case Left(error)  => throw error
-      case Right(value) => value
+  def toSmithyModel: Model = {
```

Comment:
**question:** has it been considered that we'd convert to the `dynamic.Model` representation instead? That one can still be loaded into a `Model.assembler` and serialized to the IDL, but it'd be a cross-platform representation.